Simbody/include/simbody/internal/SimbodyMatterSubsystem.h

```diff
@@ -1895,6 +1895,146 @@ void calcP(const State& state, Matrix& P) const;
+active acceleration-level constraint equations associated with holonomic and
+non-holonomic constraints, respectively, n is the number of mobilities. This is
+an O(mp+mv+n) operation.
+
+If you are going to call this method repeatedly at the same time, positions and
+velocities, you should precalculate the bias term once and supply it to the
+alternate signature of this method. See the Implementation section for more
+information.
+
+@pre \a state realized to Velocity stage
+@par Implementation
+This is accomplished by treating the input vector \a ulike as though it were
+a set of generalized accelerations (for non-holonomic constraints) or
+generalized speeds (for holonomic constraints). These are mapped to body
+accelerations (or velocities) in O(n) time. See
+calcBodyAccelerationFromUDot() for more information (converting from
+generalized speeds to velocities is just multiplying by the System Jacobian).
+The method calcBiasForMultiplyByPV() is used to determine the state-dependent
+term of the constraint error equations. Then a second call is made to
+evaluate the bias terms paerr(t,q,u;0)=-bp(t,q,u) and vaerr(t,q,u;0)=-bv(t,q,u).
+We then calculate PVulike = [Pulike; Vulike] in O(mp+mv) time, where
+Pulike = paerr(t,q,u;ulike)-paerr(t,q,u;0) and
+Vulike = vaerr(t,q,u;ulike)-vaerr(t,q,u;0).
+
+@see calcBiasForMultiplyByPV() **/
+void multiplyByPV(const State&  state,
+                  const Vector& ulike,
+                  Vector&       PVulike) const {
+    Vector biaspv;
+    calcBiasForMultiplyByPV(state, biaspv);
+    multiplyByPV(state, ulike, biaspv, PVulike);
+}
+
+/** Multiply PVulike=PV*ulike where PV = [P;V] using the supplied precalculated
+bias vector to improve performance (approximately 2X) over the other signature.
+@see calcBiasForMultiplyByPV() **/
+void multiplyByPV(const State&   state,
+                  const Vector&  ulike,
+                  const Vector&  biaspv,
+                  Vector&        PVulike) const;
+
+/** Calculate the bias vector needed for the higher-performance signature of
+the multiplyByPV() method above.
+
+@param[in]      state
+    Provides time t, positions q, and speeds u; must be realized through
+    Velocity stage so that all body spatial velocities are known.
+@param[out]     bias
+    This is the bias vector for use in repeated calls to multiplyByPV(). It
+    will be resized if necessary to length mp+mv, the total number of
+    active acceleration-level equations for the holonomic and non-holonomic
+    constraints.
+
+@pre \a state realized to Velocity stage
+@par Implementation
+This method uses either velocity- or acceleration- level constraint error
+functions with zero input to determine the bias term for use in
+multiplyByPV(). Body quantities and generalized quantities are supplied to each
+of the m active constraints' (constant time) error methods to calculate
+<pre>
+   pverr(t,q,u;ulike)=P*ulike - c(t,q)     (holonomic)
```

Comment:
I based this equation on the docs above for `calcBiasForMultiplyByG` above, but per the Simbody theory manual it seems to me that this should be either
```
pverr(t,q;qlike)=Pq*qlike - c(t,q)
```
or
```
pverr(t,q,u;ulike)=P*ulike - b_p(t,q,u)
```

Or are they interchangeable if `ulike = 0` (or `qlike = 0`)?